src/axom/core/tests/core_map.hpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/core/StackArray.hpp" /* for axom::StackArray */
+#include "axom/core/Map.hpp"
+#include "gtest/gtest.h"            /* for TEST and EXPECT_* macros */
+#include <string>
+
+namespace axom
+{
+
+namespace internal
+{
+
+template <typename Key, typename T, int N>
+Map <Key, T>  init()
+{
+  Map<Key, T> test(N/2, 2);
+  return test;
+
+}
+
+template <typename Key, typename T, int N>
+void  test_storage(Map <Key, T> &test)
+{
+  
+  for(int i = 0; i < N; i++)
+    {
+      test.insert(i, i*27-N/2);
+    }
+  
+  for(int i = 0; i < N; i++){
+    EXPECT_EQ(i*27-N/2, test.find(i).value);
+  }
+  auto ret = test.insert(N, 900);
```

Comment:
My reading is that a phantom insertion doesn't exist by default for unordered_map, but wouldn't be too terrible to implement -- simply a matter of skipping the step of actual assignment. 

Since it requires a name, perhaps, well, phantom_insert? test_insert?

Reply:
I'm not a fan of `phantom_insert` (personally).
Some other possibilities: 
* `can_insert` 
* `check_insert`
* <not having this function>

Reply:
I think 'can' if it returns a boolean, 'check' if it returns something more complex?